lib/apr/views/commerce/commerce_order_slack_view.ex

```diff
@@ -8,15 +8,25 @@ defmodule Apr.Views.CommerceOrderSlackView do
 
   alias Apr.Subscriptions.Subscription
 
-  def render(
-        %Subscription{theme: "fraud"},
-        %{"verb" => verb, "properties" => %{"items_total_cents" => items_total_cents, "mode" => mode}},
-        _routing_key
-      )
-      when items_total_cents < 3_000_00 or verb != "submitted" or mode == "offer",
-      do: nil
+  def render(nil, event, routing_key) do
+    generate_slack_message(event, routing_key)
+  end
+  
+  def render(subscription, event, routing_key) do
+    case {subscription.theme, event["verb"], event["properties"]} do
+      {"fraud", "submitted", %{"mode" => "buy", "items_total_cents" => cents}} when cents >= 3_000_00 ->        
+        generate_slack_message(event, routing_key)
+      {"fraud", "approved", %{"mode" => "offer", "currency_code" => currency_code, "items_total_cents" => cents}} 
+        when cents >= 9_500_00 and currency_code == "EUR" or currency_code == "GBP" ->
+        generate_slack_message(event, routing_key)
+        # When subscription theme is not fraud it is nil, in this case we want to render all the messages
+      {nil, _, _} ->
```

Comment:
Curious if this is already got pattern matched above in
```elixir
def render(nil, event, routing_key) do
```

Reply:
The match on line 23 covers the case when a `subscription.theme` is `nil` which is the case for all non-fraud themed subscriptions.

The `def render(nil, event, routing_key) do` handles the case when the `subscription` itself is `nil` which is possible and was captured in a pre-existing test case.

Reply:
Got it. Thanks for explaining! Yeah, I'm also not sure when we want to pass `nil` as the `subscription` here. 🤔